int32 index
uint8 r
uint8 g
uint8 b
float32 brightness # -1.0 to keep current per-pixel brightness
---
bool ok
string message
